### PR TITLE
Base module

### PR DESCRIPTION
**Pull request by DzmitryFomchyn**
### Description
This PR adds a base module and moves core logic to the `base`.

### Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have updated the `CHANGELOG` including this PR (where applicable)
- [x] I have run tests and automatic checks locally
- [ ] I have run `pitest` check locally and checked that the coverage increased (or didn't change) or decreased insignificantly
- [x] I have made corresponding changes to the documentation (where applicable)
- [x] I have grouped commits logically or I promise to squash my commits before merge

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -63,11 +63,13 @@ commands:
           name: Verify codebase
           command: |
             cd MapboxSearch
+            ./gradlew :base:lint
             ./gradlew :sdk:lint
             ./gradlew :ui:lint
             ./gradlew :sdk-common:lint
             ./gradlew :autofill:lint
             ./gradlew :sample:lint
+            ./gradlew :base:lint
```
Comment (globaltrouble) [inline .circleci/config.yml:72]: why do we need to run same command again?

Comment (DzmitryFomchyn) [reply .circleci/config.yml:72]: Thank you, great catch! I'll fix this.

**Comment by DzmitryFomchyn** [comment]
> Could you please provide a list of changes/files that require review, because it is really tidy to review 265 files?

@globaltrouble you were assigned automatically, so I think you can skip this PR because there're mostly Android-specific code. We've agreed with @sarochych on files that must be reviewed by him because of similar functionality on Android/iOS.

**Review by kalbaxa — APPROVED** [review]
Approve the extraction of Base module.

**Pull request merged**